### PR TITLE
docs: adding stack block

### PR DESCRIPTION
**Pull request by AnnuCode**

### CODE REVIEW

File: cli/reference/blocks/stack.md [hunk 1]
```diff
@@ -4,3 +4,44 @@ description: Learn how to configure stacks using the stack block.
...
 
 # The `stack` block
+
+Use the `stack` block to define a stack within a stack directory. Terramate detects stacks based on this `stack` block, and you can define only one `stack` block per stack directory. The stack metadata defined in the `stack` block is accessible only from the stack directory where it is defined. You can access this metadata via the `terramate` namespace using the `terramate.stack` object. For example, `terramate.stack.<key>` where `<key>` is any key defined in the `stack` block.
+
+## Arguments
+
+You can see `stack` metadata details [here](../../stacks/configuration.md#general-stack-metadata).
+
+- `id` (string) The user-defined id of the stack.
```
Comment (soerenmartius) [inline cli/reference/blocks/stack.md:14]: Seems like we are missing arguments such as

- `before`
- `after`
- `wants`
- `wanted_by`
- `watch`

Are missing here - for details, please see https://terramate.io/docs/cli/stacks/configuration#after

**Review by soerenmartius — CHANGES_REQUESTED** [review]

File: cli/reference/blocks/stack.md [hunk 2]
```diff
@@ -4,3 +4,44 @@ description: Learn how to configure stacks using the stack block.
 ---
 
 # The `stack` block
+
+Use the `stack` block to define a stack within a stack directory. Terramate detects stacks based on this `stack` block, and you can define only one `stack` block per stack directory. The stack metadata defined in the `stack` block is accessible only from the stack directory where it is defined. You can access this metadata via the `terramate` namespace using the `terramate.stack` object. For example, `terramate.stack.<key>` where `<key>` is any key defined in the `stack` block.
```
Comment (soerenmartius) [inline cli/reference/blocks/stack.md:8]:
```suggestion
Use the `stack` block to define a stack within a stack directory. Terramate detects stacks based on this `stack` block, and you can define only one `stack` block per stack directory. Use the `stack` block to configure:

- The metadata of stack
- The orchestration behavior of a stack
- The change detection behavior of a stack

## Metadata

The stack metadata defined in the `stack` block is accessible only from the stack directory where it is defined. You can access this metadata via the `terramate` namespace using the `terramate.stack` object. For example, `terramate.stack.<key>` where `<key>` is any key defined in the `stack` block.

## Orchestration Behavior

... tbc

## Change Detection Behavior

... tbc
```

**Review by soerenmartius — COMMENTED** [review]
just sharing some more ideas here.. basically the `stack` block does three things:

- Configuring metadata
- configuring orchestration behavior
- Configurting change detection behavior

File: cli/reference/blocks/stack.md [hunk 3]
```diff
@@ -4,3 +4,70 @@ description: Learn how to configure stacks using the stack block.
 ---
 
 # The `stack` block
+
+Use the `stack` block to define a stack within a stack directory. Terramate detects stacks based on this `stack` block, and you can define only one `stack` block per stack directory. Use the `stack` block to configure:
```
Comment (snakster) [inline cli/reference/blocks/stack.md:8]:
```suggestion
Use the `stack` block to define a stack within a directory. Terramate detects stacks based on this `stack` block, and you can define only one `stack` block per stack directory. Use the `stack` block to configure:
```

File: cli/reference/blocks/stack.md [hunk 4]
```diff
@@ -4,3 +4,70 @@ description: Learn how to configure stacks using the stack block.
...
+- The metadata of stack
+- The orchestration behavior of a stack
+- The change detection behavior of a stack
+
+### Metadata
+
+The stack metadata defined in the `stack` block is accessible only from the stack directory where it is defined. You can access this metadata via the `terramate` namespace using the `terramate.stack` object. For example, `terramate.stack.<key>` where `<key>` is any key defined in the `stack` block.
```
Comment (snakster) [inline cli/reference/blocks/stack.md:15]: The `accessible` could maybe use some leading context. In the context of terramate run? For globals? Unless this is part of a bigger picture and the context is obvious, but to me looking at this part only, it's not.

**Review by snakster — COMMENTED** [review]
Looks good overall 👍 
I left some minor comments.

File: cli/reference/blocks/stack.md (hunk 4 above)
Comment (AnnuCode) [reply cli/reference/blocks/stack.md]: I replaced 'accessible' with 'scoped locally'. Will that be good?

File: cli/reference/blocks/stack.md [hunk 5]
```diff
@@ -4,3 +4,71 @@ description: Learn how to configure stacks using the stack block.
...
+- The orchestration behaviour of a stack
+- The change detection behaviour of a stack
+
+### Metadata
+
+The stack metadata defined in the `stack` block is scoped locally to the directory where it is defined. You can access this metadata via the `terramate` namespace using the `terramate.stack` object. For example, `terramate.stack.<key>` where `<key>` is any key defined in the `stack` block.
+
+### Orchestration Behavior
+
+In Terramate, the default execution order is that the parent stacks always run before child stacks as per the filesystem hierarchy. For a particular stack, you can override this default and control which stacks will run before or after by explicitly defining the execution order in the `stack` block using the `before`, `after`, or `wants` keywords. For details, please see [orchestration](../../orchestration/index.md).
```
Comment (soerenmartius) [inline cli/reference/blocks/stack.md:19]:
```suggestion
In Terramate, the [default execution order](../../orchestration/index.md#default-order-of-execution) is that the parent stacks always run before child stacks as per the filesystem hierarchy. For a particular stack, you can override this default and control which stacks will run before or after by explicitly defining the execution order in the `stack` block using the `before`, `after`, or `wants` keywords. For details, please see [orchestration](../../orchestration/index.md).
```

File: cli/reference/blocks/stack.md [hunk 6]
```diff
@@ -4,3 +4,71 @@ description: Learn how to configure stacks using the stack block.
...
+|------------------|----------------|-------------|
+| id               | string         | The id of the stack |
+| name             | string         | The name of the stack |
+| description      | string         | The description of the stack |
+| tags             | list(string)   | The tags of the stack |
+| before           | list(string)   | Always run this stack [`before`](../../stacks/configuration.md#before) stacks in this list |
+| after            | list(string)   | Always run this stack [`after`](../../stacks/configuration.md#after) stacks in this list |
+| wants            | list(string)   | Always run these [`wanted`](../../stacks//configuration.md#wants) stacks when this stack is run |
+| wanted_by            | list(string)   | Always run this stack when running stacks in this list as it is [`wanted_by`](../../stacks//configuration.md#wanted_by) them |
+| watch            | list(string)   | Mark this stack as changed if any files in this [`watch`](../../stacks/configuration.md#watch) list is changed
```
Comment (soerenmartius) [inline cli/reference/blocks/stack.md:40]: What's missing here is what values are accepted by `before`and `after` (e.g. relative path, absolute path, tags, etc.)

Comment (soerenmartius) [reply cli/reference/blocks/stack.md:40]: I would elaborate on this to make sure we document all values accepted

Comment (soerenmartius) [reply cli/reference/blocks/stack.md:40]: Similar for `watch` btw which accepts:

- the file path can be relative to the stack
- the file path can be project absolute
- the file must be committed to watch mechanism work
- the watch expression supports funcalls but no variable

Comment (soerenmartius) [reply cli/reference/blocks/stack.md:40]: @AnnuCode perhaps the best is to jump on a quick call with @i4ki for getting those right

File: cli/reference/blocks/stack.md [hunk 7]
```diff
@@ -4,3 +4,71 @@ description: Learn how to configure stacks using the stack block.
...
+
+
+## Syntax
+
+```hcl
+stack{
```
Comment (soerenmartius) [inline cli/reference/blocks/stack.md:46]: indentation is off here.. please always format all examples using `terramate fmt`

Comment (soerenmartius) [inline cli/reference/blocks/stack.md:46]:
```suggestion
stack {
```

File: cli/reference/blocks/stack.md [hunk 8]
```diff
@@ -4,3 +4,71 @@ description: Learn how to configure stacks using the stack block.
...
+    id = "some-id"
+    name = "name can be different from stack.tm.hcl"
```
Comment (soerenmartius) [inline cli/reference/blocks/stack.md:48]:
```suggestion
    name = "Some Name"
```

File: cli/reference/blocks/stack.md [hunk 9]
```diff
@@ -4,3 +4,71 @@ description: Learn how to configure stacks using the stack block.
...
+### Arguments
+
+You can see `stack` argument details [here](../../stacks/configuration.md#general-stack-metadata).
+
+| name             |      type      | description |
+|------------------|----------------|-------------|
+| id               | string         | The id of the stack |
+| name             | string         | The name of the stack |
+| description      | string         | The description of the stack |
+| tags             | list(string)   | The tags of the stack |
```
Comment (soerenmartius) [inline cli/reference/blocks/stack.md:35]: all three values are synced with Terramate Cloud and help you to keep track of changes in Terramate Cloud btw

**Review by soerenmartius — CHANGES_REQUESTED** [review]

File: cli/reference/blocks/stack.md [hunk 10]
```diff
@@ -4,3 +4,71 @@ description: Learn how to configure stacks using the stack block.
...
+    "aws",
+    "vpc",
+    "bastion",
+  ]
+   watch = [
+    "/external/file1.txt",
+    "/external/file2.txt"
+   ]
+  after = [
+  "tag:prod:networking",
```
Comment (i4ki) [inline cli/reference/blocks/stack.md:70]:
```suggestion
  "tag:prod:networking", # all stacks containing both tags `prod` and `networking`.
```
Maybe a side comment would be nice here but I'm not sure, it's up to you guys.

**Review by i4ki — APPROVED** [review]
LGTM but I left some small improvements.
Not sure if they makes sense.

**Review by soerenmartius — DISMISSED** [review]

**Pull request merged**